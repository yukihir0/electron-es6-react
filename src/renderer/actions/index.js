import axios from "axios";

export function increment() {
  return { type: "INCREMENT" };
}
 
export function decrement() {
  return { type: "DECREMENT" };
}

export function asyncIncrement() {
  return dispatch => {
    setTimeout(() => {
      dispatch({type: "ASYNC_INCREMENT"});
    }, 1000);
  };
}

export function asyncDecrement() {
  return dispatch => {
    setTimeout(() => {
      dispatch({type: "ASYNC_DECREMENT"});
    }, 1000);
  };
}

export function requestItem() {
  const url = "https://qiita.com/api/v2/items";

  return dispatch => {
    axios.get(url).then(res => {
      dispatch({type: "REQUEST_ITEM", items: res.data});
    }).catch(error => {
      dispatch({type: "REQUEST_ITEM", items: []});
    });
  };
}
