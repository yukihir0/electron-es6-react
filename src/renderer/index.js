import React from "react";
import ReactDOM from "react-dom";
import {createStore, applyMiddleware} from "redux";
import {Provider} from "react-redux";
import thunk from "redux-thunk";
import Hello from "./containers/hello";
import Counter from "./containers/counter";
import AsyncCounter from "./containers/async_counter";
import QiitaButton from "./containers/qiita_button";
import reducer from "./reducers";

import "bootstrap";
import "./index.scss";

const store = createStore(
  reducer,
  applyMiddleware(thunk)
);
const rootEl = document.getElementById("root");
 
ReactDOM.render(
    <Provider store={store}>
      <Hello />
      <Counter />
      <AsyncCounter />
      <QiitaButton />
    </Provider>,
    rootEl
);
