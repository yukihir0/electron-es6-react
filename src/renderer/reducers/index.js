const initState = {
  value: 0,
  async_value: 0,
  items: []
};

export default (state = initState, action) => {
  switch (action.type) {
    case "INCREMENT":
      return Object.assign({}, state, {
        value: state.value + 1
      });
    case "DECREMENT":
      return Object.assign({}, state, {
          value: state.value - 1
      });
    case "ASYNC_INCREMENT":
      return Object.assign({}, state, {
        async_value: state.async_value + 1
      });
    case "ASYNC_DECREMENT":
      return Object.assign({}, state, {
          async_value: state.async_value - 1
      });
    case "REQUEST_ITEM":
      return Object.assign({}, state, {
        items: action.items
      });
    default:
      return state;
  }
};
