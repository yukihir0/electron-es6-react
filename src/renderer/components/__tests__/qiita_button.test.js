import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import QiitaButton from "../qiita_button";

test("should render items", () => {
  const props = {
    items: [
      { title: "hoge" },
      { title: "fuga" }
    ]
  };
  const element = shallow(
    <QiitaButton{...props} />
  );

  expect(element.find("ul").html()).toBe("<ul><li>hoge</li><li>fuga</li></ul>");
});

test("if button click, should call requestItem function", () => {
  const mockFn= jest.fn();
  const props = {
    items: [],
    onClickQiitaButton: mockFn,
  };
  const element = shallow(
    <QiitaButton{...props} />
  );

  element.find("#btn-qiita").simulate("click");
  expect(mockFn).toHaveBeenCalled();
});

test("snapshot", () => {
  const props = {
    items: []
  };
  const element = renderer.create(<QiitaButton{...props} />).toJSON();

  expect(element).toMatchSnapshot();
});
