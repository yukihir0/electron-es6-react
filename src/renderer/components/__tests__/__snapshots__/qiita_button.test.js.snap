// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <p>
      <div
        className="col"
      >
        <button
          className="col btn btn-outline-primary btn-block"
          id="btn-qiita"
          type="button"
        >
          request
        </button>
        <ul />
      </div>
    </p>
  </div>
</div>
`;
